api/src/test/java/org/openmrs/module/dhisreport/api/impl/DHIS2ReportingServiceImplTest.java

```diff
@@ -0,0 +1,277 @@
+        {
+            DataValueSet dvs = dhis2ReportingService.evaluateReportDefinition( rd, period, location );
+            List<DataValue> datavalues = dvs.getDataValues();
+            for ( DataValue dv : datavalues )
+            {
+                assertEquals( "3", dv.getValue() );
+            }
+
+        }
+
+    }
+
+    @Test
+    public void shouldGenerateReportingReportDefinition()
+        throws Exception
+    {
+        ClassPathResource resource = new ClassPathResource( "reportDefinition.xml" );
+        dhis2ReportingService.unMarshallandSaveReportTemplates( resource.getInputStream() );
+        ReportTemplates rt = dhis2ReportingService.getReportTemplates();
+        List<ReportDefinition> lt = rt.getReportDefinitions();
+        for ( ReportDefinition l : lt )
+        {
+            l.setReportingReportId( "rt123" );
+        }
+        dhis2ReportingService.saveReportTemplates( rt );
+        String timeperiod = "2016-06-28";
+        Location location = new Location();
+        location.setName( "County General" );
+        location.setDescription( "desc" );
+        location.setAddress1( "address1" );
+        location.setId( 123 );
+        Period period = new MonthlyPeriod( new SimpleDateFormat( "yyyy-MM-dd" ).parse( timeperiod ) );
+        SimpleDateFormat dateformat = new SimpleDateFormat( "yyyy-MM-dd" );
+        String start = "2016-06-28";
+        String end = "2016-06-30";
+        Date startDate = null;
+        Date endDate = null;
+
+        startDate = dateformat.parse( start );
+        endDate = dateformat.parse( end );
+
+        Map<String, Mapped<? extends DataSetDefinition>> mp = new HashMap<String, Mapped<? extends DataSetDefinition>>();
+        SimplePatientDataSetDefinition maleView = new SimplePatientDataSetDefinition();
+        maleView.addPatientProperty( "patientId" );
+        mp.put( "1", new Mapped<DataSetDefinition>( maleView, null ) );
+        SimplePatientDataSetDefinition femaleView = new SimplePatientDataSetDefinition();
+        femaleView.addPatientProperty( "patientId" );
+        mp.put( "2", new Mapped<DataSetDefinition>( femaleView, null ) );
+
+        period.setStartDate( startDate );
+        period.setEndDate( endDate );
+        for ( ReportDefinition rd : lt )
+        {
+
+            ReportDefinitionService service = Context.getService( ReportDefinitionService.class );
+            PeriodIndicatorReportDefinition report = new PeriodIndicatorReportDefinition();
+            report.setName( rd.getName() );
+            report.setId( rd.getId() );
+            report.setUuid( rd.getReportingReportId() );
+            report.setDataSetDefinitions( mp );
+            org.openmrs.module.reporting.report.definition.ReportDefinition savedReportDefinition = service
+                .saveDefinition( report );
+
+            // DataValueSet dvs =
+            // dhis2ReportingService.generateReportingReportDefinition(rd,
+            // period, location);
+
```

Comment:
@maurya The solution of adding dataset as mentioned [here](https://github.com/openmrs/openmrs-module-reporting/blob/e8a024e8762bb795094dbb65b3c912e37a45b888/api-tests/src/test/java/org/openmrs/module/reporting/data/patient/evaluator/ConvertedPatientDataEvaluatorTest.java) didn't work. The error still persist. Any clue about how we can proceed.